### ISSUE TITLE
Use type _doc instead of doc for elasticsearch >= 7.0

### ISSUE THREAD

**Issue by graphaelli** [elastic/beats#7612]
With the [removal of mapping types](https://www.elastic.co/guide/en/elasticsearch/reference/current/removal-of-types.html) coming in the next major release, events should use the reserved `_doc` type.

Since logstash has already updated for 7.0, mixing beats using the elasticsearch and logstash outputs will trigger as error:

    Rejecting mapping update to [xbeat-7.0.0-alpha1-2018.07.13] as the final mapping would have more than 1 type: [_doc, doc]

**Comment by ruflin** [comment]
Agree we should migrate but I wonder for the above case, if actually Logstash should take the type communicated by Beats?

**Comment by djp-search** [comment]
In the interim elasticsearch < 7.0 it might be useful to allow the option to set _type => _doc to help users with the transition for architectures where beats talk direct to ingest nodes

This could be achieved e.g. with a simple pipeline

```
PUT _ingest/pipeline/doc2_doc
{
  "processors": [
    {
      "script": {
        "source": """ctx._type= "_doc";""",
        "lang": "painless"
      }
    }
  ]
}

# Test 

POST _ingest/pipeline/doc2_doc/_simulate
{
  "docs": [
    {
      "_id": "typedoc",
      "_type": "doc",
      "_source": {}
    },
    {
      "_id": "type_doc",
      "_type": "_doc",
      "_source": {}
    },
    {
      "_id": "typenull",
      "_type": null,
      "_source": {}
    },
    {
      "_id": "typeundefined",
      "_source": {}
    }
  ]
}

# Expected output

{
  "docs": [
    {
      "doc": {
        "_index": "_index",
        "_type": "_doc",
        "_id": "typedoc",
        "_source": {},
        "_ingest": {
          "timestamp": "2018-07-17T09:21:33.631Z"
        }
      }
    },
    {
      "doc": {
        "_index": "_index",
        "_type": "_doc",
        "_id": "type_doc",
        "_source": {},
        "_ingest": {
          "timestamp": "2018-07-17T09:21:33.631Z"
        }
      }
    },
    {
      "doc": {
        "_index": "_index",
        "_type": "_doc",
        "_id": "typenull",
        "_source": {},
        "_ingest": {
          "timestamp": "2018-07-17T09:21:33.631Z"
        }
      }
    },
    {
      "doc": {
        "_index": "_index",
        "_type": "_doc",
        "_id": "typeundefined",
        "_source": {},
        "_ingest": {
          "timestamp": "2018-07-17T09:21:33.631Z"
        }
      }
    }
  ]
}
```

Would it be preferable to implement this by predefining the pipeline in Elasticsearch and using the "pipeline:" parameter in <beat>.yml or better upstream in the beat itself?

Putting the pipeline in Elasticsearch is the minimum impact solution, my concern here would be to establish a pattern which might not be desirable in the future

**Comment by graphaelli** [comment]
closed by #9056
